### DISCUSSION TITLE
chown during update is not cool... :(

### DISCUSSION THREAD

**Discussion by cmh714** [rern/rAudio#904] [General]
I have 6TB of files, >105k and my NAS are RO so the chown fails anyways.....and now the webpage updating crashed. Going to just be patient and see if it comes back as updated in a few hours. Refreshing the UI tells me I need to update....but going to wait.

**Comment by somax80** [comment]
After upgrade my mpd daemon fails to start with message "permission denied on /srv/http/data/mpd/mpd.log"
Permissions of all /srv/http changed in 755 and owner changed in http:http (except mpd.db file that has owner mpd)
I changed the permissions of log file, the daemon started,  but i've got no more library!
Mpd daemon run as "mpd" user, but all directory /srv/http has owner "http" and permissions 755.
The script srv/http/bash/cmd.sh contains:
`chmod -R 755 /srv/http
	chown -R http:http /srv/http
	chown -R mpd:audio $dirplaylists /mnt/MPD
	chown mpd:audio $dirmpd/mpd.db
	;;`

Could this be the problem?

Thanks.

**Comment by ceejayemm** [comment]
@rern 

I have to agree with the others on this list.  You can't go changing the permissions, in my case on the remote NFS share which holds my music library (/mnt/MPD/NAS/NAS_Music). For one thing it takes an age and for another it totally destroys the ownership of the files on the remote share which might well be used by other, non-rAudio-1, services.  The issue is caused by:

chown -R mpd:audio $dirplaylists /mnt/MPD

(in the dirpermissions function of /srv/http/bash/cmd.sh)

as my Music Library NFS mount is soft linked (by rAudio-1) as /mnt/MPD/NAS/NAS_Music

I haven't seen the "permission denied on /srv/http/data/mpd/mpd.log" that @somax80 has reported nor "the webpage updating crash" reported by @cmh714 BUT I have seen the time delay to the update and, more importantly, the changing of the permissions on my NFS music library share.

Chris

**Comment by cmh714** [comment]
I waited over 6 hours to hopefully let it finish since I lost the Updating UI. I was able to do a reboot and things seemed to come back fine. However, since I do not have the r1 file, the Update page says to download a new image. I jammed the date into the file, so will wait for the next update to see if this is fixed....

On another note, I was think a possible way around this is to unmount the NAS perform the update and then remount the NAS.....with the hope that it wont want to chown 6TB.

**Comment by rern** [comment]
Fixed with today's update.

**Reply by cmh714** [reply]
As always, THANK YOU SIR!

**Reply by somax80** [reply]
> Fixed with today's update.

Thank you!

**Comment by ceejayemm** [comment]
Hi

This seems better after update 20220428, the delay to the update completing has gone and the NAS Share is (mainly) not having its ownership changed.  The one exception is the database held in the shareddata area:

rAudio-1:~  #  ls -lrt /srv/http/data/mpd
lrwxrwxrwx 1 http http 24 Apr 24 19:43 /srv/http/data/mpd -> /srv/http/shareddata/mpd

rAudio-1:/srv/http/data/mpd  #  ls -lrt
total 324
-rwxrwxr-x+ 1 mpd  audio 179438 Feb 24 08:30 mpd.db
-rwxrwxr-x+ 1 root   100    194 Feb 24 08:30 mpdstate
-rwxrwxr-x+ 1 root   100    118 Feb 24 08:51 mpdignorelist
-rwxrwxr-x+ 1 root   100    213 Mar 23 09:46 counts
-rwxrwxr-x+ 1 root   100  28860 Mar 29 08:24 albumbyartist
-rwxrwxr-x+ 1 root   100  28860 Mar 29 08:24 album
-rwxrwxr-x+ 1 root   100   2543 Mar 29 08:24 albumartist
-rwxrwxr-x+ 1 root   100  16719 Mar 29 08:24 artist
-rwxrwxr-x+ 1 root   100  33588 Mar 29 08:24 composer
-rwxrwxr-x+ 1 root   100     36 Mar 29 08:24 conductor
-rwxrwxr-x+ 1 root   100    534 Mar 29 08:24 genre
-rwxrwxr-x+ 1 root   100   2759 Mar 29 08:24 date

Chris
